src/Illuminate/Database/Eloquent/Concerns/HasAttributes.php

```diff
@@ -426,6 +427,10 @@ protected function getAttributeFromArray($key)
      */
     public function getRelationValue($key)
     {
+        if ($this->strictLoading && ! $this->relationLoaded($key)) {
```

Comment:
Another alternative may be moving the condition here and avoid the changes in the Eloquent builder:
```suggestion
        if ($this->getConnection() && $this->getConnection()->getConfig('strict_load') && ! $this->relationLoaded($key)) {
```